implement_me.py

```diff
@@ -0,0 +1,115 @@
+from dateutil import parser
+from elasticsearch.helpers import bulk
+from elastic_handler import get_elastic_client
+
+INDEX_NAME = 'entry-index'
+ENTRY_TYPE = 'ip_entry'
+MAX_DOCUMENTS = 100000
+BULK_SIZE = 1000
+
+es = get_elastic_client()
+
+
+def clear_index():
+    es.indices.delete(index=INDEX_NAME, ignore=[400, 404])
+    init_mapping()
+
+
+def init_mapping():
+    index_mapping = """{
+        "mappings": {
+            "ip_entry":
+                {
+                  "properties": {
+                    "ip": { 
+                      "type":     "text",
+                      "fielddata": true
+                    }
+                  }
+                }
+        }
+    }"""
+    if not es.indices.exists(INDEX_NAME):
+        es.indices.create(INDEX_NAME, index_mapping)
+
+
+def index(data):
+    """
+    Index the given entries.
+
+    :note: May be called multiple times.
+
+    :param data: A list of 'Entry' instances to index.
+    """
+    indexes = []
+    for entry in data:
+        doc = dict(entry._asdict())
+        doc['_type'] = ENTRY_TYPE
+        doc['_index'] = INDEX_NAME
+        indexes.append(doc)
+        # Index in bulks.
+        if len(indexes) == BULK_SIZE:
+            bulk(es, indexes)
+            indexes.clear()
+
+    bulk(es, indexes)
+    es.indices.flush()
+
+
+def get_device_histogram(ip, n):
+    """
+    Return the latest 'n' entries for the given 'ip'.
+    """
+    query = {
+        'match': {'ip': ip}
+    }
+
+    body = {
+        'from': 0,
```

Comment:
`from` is not needed in this case (0 is the default).

Reply:
OK